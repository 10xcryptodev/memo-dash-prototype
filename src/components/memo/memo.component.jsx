import React, { Component } from 'react'
import { Comment, Segment } from 'semantic-ui-react'
import MemoActionsContainer from './actions/memo-actions.container'
import MemoAvatarContainer from './avatar/memo-avatar.container'
import MemoContentContainer from './content/memo-content.container'
import './memo.styles.css'
import MemoContainer from './memo.container'
import MemoDeleteContainer from './delete/memo-delete.container'

export default class MemoComponent extends Component {
  componentDidMount() {
    const { memo, showReplies, onLoadReplies } = this.props
    if (showReplies && memo.memoRepliesCount > 0) {
      onLoadReplies(memo)
    }
  }

  componentWillReceiveProps(nextProps) {
    const { memo, closeModal } = nextProps

    if (!memo) closeModal()
  }

  render() {
    const {
      memo,
      showReplies,
      showDelete,
      showBorders,
      replies,
      openModalOnClick,
      onModalOpenClicked
    } = this.props

    let {memoNotAvailable} = this.props

    if (memo === undefined) {
      return null
    }

    return (
      <Segment
        basic={!showBorders}
        className="memo"
        onClick={!!openModalOnClick && !memoNotAvailable ? () => onModalOpenClicked(memo) : undefined}
        padded
      >
        {memoNotAvailable ? (
          <Comment>
            <Comment.Content>Liked memo is no longer available</Comment.Content>
          </Comment>
        ) : (
          <Comment>
            <MemoAvatarContainer memo={memo} />
            <Comment.Content>
              {showDelete ? <MemoDeleteContainer memo={memo} /> : null}
              <MemoContentContainer memo={memo} />
              <MemoActionsContainer memo={memo} />
            </Comment.Content>
            {showReplies && !!replies && (
              <Comment.Group>
                {replies.map((reply, k) => (
                  <MemoContainer showReplies={true} memo={reply} key={k} />
                ))}
              </Comment.Group>
            )}
          </Comment>
        )}
      </Segment>
    )
  }
}
