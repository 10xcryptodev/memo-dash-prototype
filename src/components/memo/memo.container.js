import { connect } from 'react-redux'

import MemoComponent from './memo.component'
import { openMemoModal, getMemoReplies, closeMemoModal } from '../../store/actions'
import { getMemosByCombinedIds, isMemoOfCurrentUser } from '../../store/selectors'

const mapStateToProps = (state, ownProps) => {
  const { memo, showReplies } = ownProps

  if (!memo) return {}

  let replies = undefined
  if (showReplies) {
    replies = getMemosByCombinedIds(memo.replyIds)(state)
  }

  return { replies, showDelete: isMemoOfCurrentUser(memo)(state) }
}

const mapDispatchToProps = dispatch => {
  return {
    onModalOpenClicked: memo => dispatch(openMemoModal(memo)),
    closeModal: () => dispatch(closeMemoModal()),
    onLoadReplies: memo => dispatch(getMemoReplies(memo.username, memo.idx))
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MemoComponent)
