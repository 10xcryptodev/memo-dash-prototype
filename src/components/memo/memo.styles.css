.memo {
  cursor: pointer;
}

.memo .actions > span {
  margin: 0 0.75em 0 0;
}
