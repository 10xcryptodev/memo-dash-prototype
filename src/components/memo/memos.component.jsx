import React from 'react'
import MemoContainer from './memo.container'
import { Comment, Segment } from 'semantic-ui-react'
import MemoModalContainer from './modal/memo-modal.container'

const MemosComponent = props => {
  const { memos } = props

  if(!memos){
    return null;
  }

  return (
    <Segment basic center>
      {memos ? (
        <Comment.Group>
          {Object.keys(memos)
            .sort((a, b) => a.memoDatetime < b.memoDatetime)
            .map(memoId => (
              <MemoContainer
                key={memoId}
                showBorders={true}
                openModalOnClick={true}
                showReplies={false}
                memo={memos[memoId]}
              />
            ))}
          <MemoModalContainer />
        </Comment.Group>
      ) : (
        'No memos available'
      )}
    </Segment>
  )
}

export default MemosComponent
