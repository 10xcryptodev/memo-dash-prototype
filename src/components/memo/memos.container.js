import { connect } from 'react-redux'

import MemosComponent from './memos.component'

const mapStateToProps = (state, ownProps) => {
  return {}
}

const mapDispatchToProps = dispatch => {
  return {}
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MemosComponent)
