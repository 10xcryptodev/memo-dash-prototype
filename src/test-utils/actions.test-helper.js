import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

const verifyAction = (type, payload, actionCreator) => {
  const expectedAction = {
    type,
    payload
  }
  expect(actionCreator(payload)).toEqual(expectedAction)
}

const mockStore = configureMockStore([thunk])

const mockStoreAndDispatch = async (state, actionCreator) => {
  const store = mockStore(state)
  await store.dispatch(actionCreator)
  return store.getActions()
}

const findAction = (actions, type) => actions.find(action => action.type === type)

const getAction = (actions, type) => {
  const action = findAction(actions, type)
  if (action) return Promise.resolve(action)

  return new Promise(resolve => {
    store.subscribe(() => {
      const action = findAction(store, type)
      if (action) resolve(action)
    })
  })
}

export { verifyAction, mockStoreAndDispatch, getAction }
