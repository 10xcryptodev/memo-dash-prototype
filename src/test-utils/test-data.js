export const testUsers = {
  alice: {
    username: 'alice',
    profile: {
      avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg',
      bio: 'Quod architecto molestias et.',
      followersCount: 1,
      followingCount: 1,
      likesCount: 2,
      username: 'alice'
    },
    userId: 'f0fd479ba875bf2232a4043dbefff24e2af2ca5ccd50b3ba9e91673707336726',
    likes: [
      {
        idx: 5,
        relation: {
          username: 'bob',
          userId: 'c6e5e284054bcbcd2cddea1ec36579e1fdf8e788d8a46d0351e276b4d0bb297c',
          index: 1
        }
      },
      {
        idx: 6,
        relation: {
          username: 'bob',
          userId: 'c6e5e284054bcbcd2cddea1ec36579e1fdf8e788d8a46d0351e276b4d0bb297c',
          index: 2
        }
      }
    ],
    memoIds: ['[alice][1]', '[alice][2]'],
    followers: ['bob'],
    following: ['bob']
  },
  bob: {
    username: 'bob',
    profile: {
      avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/129.jpg',
      bio: 'Id qui nisi asperiores vel quo.',
      followersCount: 1,
      followingCount: 1,
      likesCount: 1,
      username: 'bob'
    },
    userId: 'c6e5e284054bcbcd2cddea1ec36579e1fdf8e788d8a46d0351e276b4d0bb297c',
    likes: [
      {
        idx: 7,
        relation: {
          username: 'alice',
          userId: 'f0fd479ba875bf2232a4043dbefff24e2af2ca5ccd50b3ba9e91673707336726',
          index: 1
        }
      },
      {
        idx: 8,
        relation: {
          username: 'alice',
          userId: 'f0fd479ba875bf2232a4043dbefff24e2af2ca5ccd50b3ba9e91673707336726',
          index: 2
        }
      }
    ],
    memoIds: ['[bob][1]', '[bob][2]'],
    followers: ['alice'],
    following: ['alice']
  }
}

export const testMemos = {
  '[bob][1]': {
    avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/rangafangs/128.jpg',
    idx: 1,
    memoDatetime: '2018-07-26T16:45:08.218Z',
    memoLikesCount: 0,
    memoRepliesCount: 0,
    memoText: 'Qui magnam maiores laboriosam animi sit aut.',
    memoTipTotal: 0,
    username: 'bob'
  },
  '[bob][2]': {
    avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/rangafangs/128.jpg',
    idx: 2,
    memoDatetime: '2018-07-26T16:45:10.954Z',
    memoLikesCount: 0,
    memoRepliesCount: 0,
    memoText: 'Quia est odio nemo deserunt.',
    memoTipTotal: 0,
    username: 'bob'
  },
  '[alice][1]': {
    avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg',
    idx: 1,
    memoDatetime: '2018-07-26T16:44:56.000Z',
    memoLikesCount: 1,
    memoRepliesCount: 0,
    memoText: 'Suscipit vel nobis.',
    memoTipTotal: 0,
    username: 'alice'
  },
  '[alice][2]': {
    avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg',
    idx: 2,
    memoDatetime: '2018-07-26T16:44:58.620Z',
    memoLikesCount: 0,
    memoRepliesCount: 0,
    memoText: 'Praesentium libero est repudiandae et repudiandae.',
    memoTipTotal: 0,
    username: 'alice'
  }
}
