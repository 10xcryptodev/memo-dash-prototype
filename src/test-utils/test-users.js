export default [
  {
    username: 'alice',
    profile: {
      avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg',
      bio: 'Quod architecto molestias et.',
      followersCount: 1,
      followingCount: 1,
      likesCount: 1,
      username: 'alice'
    },
    ownLikes: [{ idx: 1, relation: { index: 3 } }, { idx: 2, relation: { index: 4 } }],
    memos: [
      {
        avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/chaabane_wail/128.jpg',
        idx: 1,
        memoDatetime: '2018-07-26T16:44:56.000Z',
        memoLikesCount: 1,
        memoRepliesCount: 0,
        memoText: 'Suscipit vel nobis.',
        memoTipTotal: 0,
        username: 'alice'
      },
      {
        avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/chaabane_wail/128.jpg',
        idx: 2,
        memoDatetime: '2018-07-26T16:44:58.620Z',
        memoLikesCount: 0,
        memoRepliesCount: 0,
        memoText: 'Praesentium libero est repudiandae et repudiandae.',
        memoTipTotal: 0,
        username: 'alice'
      }
    ]
  },
  {
    username: 'bob',
    profile: {
      avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/129.jpg',
      bio: 'Id qui nisi asperiores vel quo.',
      followersCount: 1,
      followingCount: 1,
      likesCount: 1,
      username: 'bob'
    },
    ownLikes: [{ idx: 1, relation: { index: 3 } }, { idx: 2, relation: { index: 4 } }],
    memos: [
      {
        avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/rangafangs/128.jpg',
        idx: 1,
        memoDatetime: '2018-07-26T16:45:08.218Z',
        memoLikesCount: 0,
        memoRepliesCount: 0,
        memoText: 'Qui magnam maiores laboriosam animi sit aut.',
        memoTipTotal: 0,
        username: 'bob'
      },
      {
        avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/rangafangs/128.jpg',
        idx: 2,
        memoDatetime: '2018-07-26T16:45:10.954Z',
        memoLikesCount: 0,
        memoRepliesCount: 0,
        memoText: 'Quia est odio nemo deserunt.',
        memoTipTotal: 0,
        username: 'bob'
      }
    ]
  }
]
