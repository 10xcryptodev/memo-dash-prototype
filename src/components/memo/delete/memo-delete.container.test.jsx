import React from 'react'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import MemoDeleteContainer from './memo-delete.container'
import { combineMemoId } from '../../../store/reducers/memo.reducer'

import { testUsers, testMemos } from '../../../test-utils/test-data'

describe('<MemoDeleteContainer />', () => {
  let store
  const alice = testUsers['alice']
  const memo = testMemos[alice.memoIds[0]]

  beforeEach(() => {
    const mockStore = configureStore()
    store = mockStore()

    const div = document.createElement('div')
    document.body.appendChild(div)
  })

  describe('Shallow rendering', () => {
    describe('should render', () => {
      it('without crashing', () => {
        const wrapper = shallow(<MemoDeleteContainer memo={memo} />, { context: { store } })
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
