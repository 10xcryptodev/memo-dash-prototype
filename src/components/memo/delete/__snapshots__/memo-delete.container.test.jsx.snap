// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoDeleteContainer /> Shallow rendering should render without crashing 1`] = `
<MemoDeleteComponent
  memo={
    Object {
      "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg",
      "idx": 1,
      "memoDatetime": "2018-07-26T16:44:56.000Z",
      "memoLikesCount": 1,
      "memoRepliesCount": 0,
      "memoText": "Suscipit vel nobis.",
      "memoTipTotal": 0,
      "username": "alice",
    }
  }
  onDeleteMemoClicked={[Function]}
/>
`;
